Public/Src/Cache/ContentStore/Distributed/NuCache/BlobCentralStorage.cs

```diff
@@ -42,22 +42,29 @@ public class BlobCentralStorage : CentralStorage
         /// <nodoc />
         public BlobCentralStorage(BlobCentralStoreConfiguration configuration)
         {
-            Contract.Requires(configuration.ConnectionStrings.Count != 0);
-
             _configuration = configuration;
 
-            _containers = _configuration.ConnectionStrings.Select(
-                (connectionString, index) =>
+            _containers = _configuration.Credentials.Select(

```

Comment:
I would create a factory method (exgternal or in Credentials) that will create CloubBlobContainer.

In this case the logic in this class will be essentially the same as it was before.

And I think it make sense to move this logic away from this type because its not essential to it. (and it'll make Credentials type easier to use).